File: src/main/java/com/github/alikemalocalan/greentunnel4jvm/log/Appender.java
package com.github.alikemalocalan.greentunnel4jvm.log;

import ch.qos.logback.classic.PatternLayout;
import ch.qos.logback.classic.spi.ILoggingEvent;
import ch.qos.logback.core.AppenderBase;
import com.github.alikemalocalan.greentunnel4jvm.gui.Gui;

/**
 * @author Rodrigo Garcia Lima (email: rodgarcialima@gmail.com | github: rodgarcialima)
 * @see ch.qos.logback.core.AppenderBase
 */
public class Appender extends AppenderBase<ILoggingEvent> {

    private PatternLayout patternLayout;

    @Override
    public void start() {
        patternLayout = new PatternLayout();
        patternLayout.setContext(getContext());
        patternLayout.setPattern("%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n");
        patternLayout.start();

        super.start();
    }

    @Override
    protected void append(ILoggingEvent event) {
        String formattedMsg = patternLayout.doLayout(event);

        Gui.getMainFrom().appendLogToGui(formattedMsg, event.getLevel());
    }
}

File: src/main/kotlin/com/github/alikemalocalan/greentunnel4jvm/logger/Appender.kt
package com.github.alikemalocalan.greentunnel4jvm.logger

import ch.qos.logback.classic.PatternLayout
import ch.qos.logback.classic.spi.ILoggingEvent
import ch.qos.logback.core.AppenderBase
import com.github.alikemalocalan.greentunnel4jvm.gui.Gui
import javax.swing.SwingUtilities

class Appender : AppenderBase<ILoggingEvent>() {
    private var patternLayout: PatternLayout? = null

    override fun start() {
        patternLayout = CustomPatternLayout(this@Appender.context)
        patternLayout?.start()
        super.start()
    }

    override fun append(event: ILoggingEvent) {
        val formattedMsg = patternLayout?.doLayout(event) ?: return
        SwingUtilities.invokeLater {
            Gui.mainForm.loggerText.appendLogToGui(formattedMsg, event.level.levelStr)
        }
    }
}